feature/appdetail/src/androidTest/java/com/merxury/blocker/feature/appdetail/AppDetailViewModelTest.kt

```diff
@@ -0,0 +1,308 @@
+    }
+
+    @Test
+    fun stateIsLoadingWhenDataAreLoading() = runTest {
+        val collectJob1 =
+            launch(UnconfinedTestDispatcher()) { viewModel.appInfoUiState.collect() }
+        val collectJob2 =
+            launch(UnconfinedTestDispatcher()) { viewModel.componentSortInfoUiState.collect() }
+
+        assertEquals(
+            Loading,
+            viewModel.appInfoUiState.value,
+        )
+        assertEquals(ComponentSortInfoUiState.Loading, viewModel.componentSortInfoUiState.value)
+
+        collectJob1.cancel()
+        collectJob2.cancel()
+    }
+
+    @Test
+    fun stateIsDefaultWhenNotUpdate() = runTest {
+        val collectJob1 =
+            launch(UnconfinedTestDispatcher()) { viewModel.componentListUiState.collect() }
+        val collectJob2 = launch(UnconfinedTestDispatcher()) { viewModel.appBarUiState.collect() }
+        val collectJob3 = launch(UnconfinedTestDispatcher()) { viewModel.tabState.collect() }
+
+        assertEquals(0, viewModel.componentListUiState.value.activity.size)
+        assertEquals(0, viewModel.componentListUiState.value.provider.size)
+        assertEquals(0, viewModel.componentListUiState.value.receiver.size)
+        assertEquals(0, viewModel.componentListUiState.value.service.size)
+        assertEquals(AppBarUiState(), viewModel.appBarUiState.value)
+        assertEquals(AppDetailTabs.Info, viewModel.tabState.value.selectedItem)
+
+        collectJob1.cancel()
+        collectJob2.cancel()
+        collectJob3.cancel()
+    }
+
+    @Test
+    fun stateIsSuccessWhenGetData() = runTest {
+        val collectJob1 =
+            launch(UnconfinedTestDispatcher()) { viewModel.appInfoUiState.collect() }
+        val collectJob2 =
+            launch(UnconfinedTestDispatcher()) { viewModel.componentSortInfoUiState.collect() }
+
+        appRepository.sendAppList(sampleAppList)
+        userDataRepository.sendUserData(sampleUserData)
+        val packageName = sampleAppList.first().packageName
+        val packageInfo = pm.getPackageInfoCompat(packageName, 0)
+        viewModel.loadAppInfo()
+        viewModel.loadComponentSortInfo()
+
+        assertEquals(
+            Success(
+                appInfo = sampleAppList.first().toAppItem(packageInfo),
+                iconBasedTheming = null,
+            ),
+            viewModel.appInfoUiState.value,
+        )
+        assertEquals(
+            ComponentSortInfoUiState.Success(
+                ComponentSortInfo(
+                    sorting = sampleUserData.componentSorting,
+                    order = sampleUserData.componentSortingOrder,
+                    priority = sampleUserData.componentShowPriority,
+                ),
+            ),
+            viewModel.componentSortInfoUiState.value,
+        )
+
+        collectJob1.cancel()
+        collectJob2.cancel()
+    }
+
+    @Test
+    fun componentSortInfoUiStateUpdateAfterChanged() = runTest {
+        val collectJob =
```

Comment:
Pay attention to code format:
```suggestion
        val collectJob = launch(UnconfinedTestDispatcher()) { 
            viewModel.componentSortInfoUiState.collect() 
        }
```